src/main/java/com/auction/retailService/service/CustomerOrderService.java

```diff
@@ -0,0 +1,197 @@
+package com.auction.retailService.service;
+
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import com.auction.retailService.constant.OrderStatusConstant;
+import com.auction.retailService.constant.RoleConstant;
+import com.auction.retailService.dto.CustomerOrderDTO;
+import com.auction.retailService.dto.CustomerOrderDetailDTO;
+import com.auction.retailService.dto.OrderPlacedDTO;
+import com.auction.retailService.dto.ProductDTO;
+import com.auction.retailService.dto.StoreProductsDTO;
+import com.auction.retailService.entity.CustomerOrder;
+import com.auction.retailService.entity.Order;
+import com.auction.retailService.entity.OrderProductsEntity;
+import com.auction.retailService.entity.StoreProduct;
+import com.auction.retailService.exception.OrderNotFoundException;
+import com.auction.retailService.exception.ProductOrderQuantityException;
+import com.auction.retailService.exception.StoreNotFoundException;
+import com.auction.retailService.exception.UserNotFoundException;
+import com.auction.retailService.exception.UserUnAuthorizedException;
+import com.auction.retailService.repository.CustomerOrderRepository;
+import com.auction.retailService.repository.OrderProductsRepository;
+
+@Service
+public class CustomerOrderService {
+	
+	@Autowired
+	CustomerOrderRepository costOrderRepositoty;
+	
+	@Autowired
+	OrderProductsRepository orderProductRepo;
+	
+	@Autowired
+	ProductService productService;
+	
+	@Autowired
+	StoreProductService storeProductservice;
+	
+	@Autowired
+	UserService userService;
+	
+	@Autowired
+	StoreService storeservice;
+	
+	
+	@Transactional
+	public OrderPlacedDTO addOrder(CustomerOrderDTO orderDto, Long orderBy) {
+		OrderPlacedDTO dto = new OrderPlacedDTO();
+		CustomerOrder order = new CustomerOrder();
+		order.setStoreId(orderDto.getStoreId());
+		order.setOrderDate(Instant.now()+"");
+		order.setComment("Order Placed");
+		order.setOrderBy(orderBy);
+		order.setOrderStatus(OrderStatusConstant.ORDERED);
+		CustomerOrder ord =  costOrderRepositoty.save(order);
+		List <OrderProductsEntity>orderDetails = orderDto.getProducts();
+		for(OrderProductsEntity entity : orderDetails) {
+			if(productService.isProductExist(entity.getProductId())) {	
+				StoreProduct storeProduct = storeProductservice.getProductQtyByStore(orderDto.getStoreId(), entity.getProductId());
+				Long prodQty = storeProduct.getQuantity();
+				if(prodQty>=entity.getQuantity()) {
+					storeProduct.setQuantity(prodQty - entity.getQuantity());
+					storeProductservice.saveStoreProductData(storeProduct);
+				}
+				else {
+					throw new ProductOrderQuantityException(orderDto.getStoreId(), prodQty, entity.getQuantity());
+				}
+				entity.setOrderId(ord.getOrderId());
+				orderProductRepo.save(entity);
```

Comment:
This has a performance problem. Set all your entities in a collection and at the end save them all if they are all valid.